lte/cloud/go/services/subscriberdb/storage/sub_proto.go

```diff
@@ -0,0 +1,335 @@
+	}
+	ret, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	if err != nil {
+		return nil, "", err
+	}
+	subProtos := ret.([]*lte_protos.SubscriberData)
+	nextPageToken, err := getNextPageToken(subProtos)
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "get next page token")
+	}
+	return subProtos, nextPageToken, nil
+}
+
+// GetByIDs returns an ordered list of subscriber protos with matching IDs.
+func (l *SubProtoStore) GetByIDs(network string, sids []string) ([]*lte_protos.SubscriberData, error) {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		rows, err := l.builder.
+			Select(subProtoProtoCol).
+			From(subProtoTableName).
+			Where(squirrel.And{
+				squirrel.Eq{subProtoNidCol: network},
+				squirrel.Eq{subProtoSidCol: sids},
+			}).
+			OrderBy(subProtoSidCol).
+			RunWith(tx).
+			Query()
+		if err != nil {
+			return nil, errors.Wrapf(err, "Get sub protos by ID for network %+v", network)
+		}
+		return parseSubProtoRows(rows)
+	}
+	ret, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	if err != nil {
+		return nil, err
+	}
+	subProtos := ret.([]*lte_protos.SubscriberData)
+	return subProtos, nil
+}
+
+// parseSubProtoRows scans db rows of serialized subscriber data, and returns a deserialized list
+// of subscriber protos.
+func parseSubProtoRows(rows *sql.Rows) ([]*lte_protos.SubscriberData, error) {
+	subProtos := []*lte_protos.SubscriberData{}
+	for rows.Next() {
+		subProtoMarshaled := []byte{}
+		err := rows.Scan(&subProtoMarshaled)
+		if err != nil {
+			return nil, errors.Wrap(err, "get sub protos from store, SQL row scan error")
+		}
+		subProto := &lte_protos.SubscriberData{}
+		err = proto.Unmarshal(subProtoMarshaled, subProto)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unmarshal sub protos from store")
+		}
+		subProtos = append(subProtos, subProto)
+	}
+	err := rows.Err()
+	if err != nil {
+		return nil, errors.Wrap(err, "get sub protos from store, SQL rows error")
+	}
+	return subProtos, nil
+}
+
+// getNextPageToken returns the next page token based on the lastIncludedEntity in the current page.
+func getNextPageToken(subProtos []*lte_protos.SubscriberData) (string, error) {
+	// The next token is empty if we have exhausted all protos in the db
+	if len(subProtos) == 0 {
+		return "", nil
+	}
+	lastSubProto := subProtos[len(subProtos)-1]
+	nextTokenUnmarshaled := &configurator_storage.EntityPageToken{
+		LastIncludedEntity: lte_protos.SidString(lastSubProto.Sid),
+	}
+	nextTokenMarshaled, err := proto.Marshal(nextTokenUnmarshaled)
+	if err != nil {
+		return "", err
+	}
+	nextToken := base64.StdEncoding.EncodeToString(nextTokenMarshaled)
+	return nextToken, nil
+}
+
+// decodePageToken returns the deserialized page token proto.
+func decodePageToken(token string) (*configurator_storage.EntityPageToken, error) {
+	marshaledToken, err := base64.StdEncoding.DecodeString(token)
+	if err != nil {
+		return nil, errors.Wrapf(err, "decode page token")
+	}
+	buf := &configurator_storage.EntityPageToken{}
```

Comment:
Add a note somewhere that we're reusing the configurator entity page token type for simplicity, but that this is technically an unrelated page token